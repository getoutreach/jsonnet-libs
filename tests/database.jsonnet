local database = import 'kubernetes/database.libsonnet';

{
    assignment: database.PostgresqlClusterServiceAssignment(
        "custom-resource-instance-name",
        app="my-app",
        namespace="my-app--bento1a"
    ) {
        bento:: "bento1a",
        personal_information:: "yes",
        database_cluster_namespace:: "shared-databases--bento1a",
        database_cluster_name:: "my-shared-database",
        team:: "fnd-qss",
        tier:: "tier-2",
    },
}
