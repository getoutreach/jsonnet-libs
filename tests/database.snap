---
apiVersion: databases.outreach.io/v1
kind: PostgresqlClusterServiceAssignment
metadata:
  annotations: {}
  labels:
    app: mydpotestservice
    name: mydpotestservice
  name: mydpotestservice
  namespace: mydpotestservice--ngb-ss2-us-east-2
spec:
  application_name: mydpotestservice
  bento: ngb-ss2-us-east-2
  database_name: mydpotestservice
  personal_information: "yes"
  postgresql_database_cluster:
    namespace: dpotestresource--ngb-ss2-us-east-2
    resource_name: dpotestresource
  resource_name: myresource
  team: fnd-qss
  tier: tier-2
---
apiVersion: databases.outreach.io/v1
kind: PostgresqlClusterServiceAssignment
metadata:
  annotations: {}
  labels:
    app: mydpotestservice
    name: mydpotestservice2
  name: mydpotestservice2
  namespace: mydpotestservice--ngb-ss2-us-east-2
spec:
  application_name: mydpotestservice
  bento: ngb-ss2-us-east-2
  database_name: mydpotestservice
  personal_information: "yes"
  postgresql_database_cluster:
    namespace: dpotestresource--ngb-ss2-us-east-2
    resource_name: dpotestresource
  resource_name: mydpotestservice
  team: fnd-qss
  tier: tier-2
---
apiVersion: databases.outreach.io/v1
kind: PostgresqlDatabaseCluster
metadata:
  annotations:
    argocd.argoproj.io/sync-wave: "-6"
  labels:
    app: dpotestresource
    name: dpotestresource
  name: dpotestresource
  namespace: dpotestresource--ngb-ss2-us-east-2
spec:
  app_name: dpotestresource
  bento: ngb-ss2-us-east-2
  cluster_parameters:
  - Name: rds.logical_replication
    Value: "1"
  - Name: wal_sender_timeout
    Value: "300000"
  - Name: wal_receiver_timeout
    Value: "300000"
  - Name: max_wal_senders
    Value: "20"
  database_name: dpotestresource
  engine:
    parameter_group_family: aurora-postgresql14
    version: "14.6"
  full_name: ""
  instance_class: db.t4g.medium
  name: dpotestresource
  personal_information: "yes"
  provisioner: AuroraRDS
  team: fnd-qss
  temp_builtin_users: true
  tier: tier-2
